src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -308,6 +329,36 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_rand() {
+        let mut ctx = EvalContext::default();
+        let arg = datum_expr(Datum::Null);
+        let op = scalar_func_expr(ScalarFuncSig::Rand, &[arg]);
+
+        let op = Expression::build(&mut ctx, op).unwrap();
+        let got = op.eval_real(&mut ctx, &[]).unwrap();
+
+        assert_eq!(got.is_some(), true);
+        assert_eq!(got.unwrap() < 1.0, true);
+        assert_eq!(got.unwrap() >= 0.0, true);
+    }
+
+    #[test]
+    fn test_rand_with_seed() {
```

Comment:
How does the following case work? (https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_rand)

 If an integer argument N is specified, it is used as the seed value:

- With a constant initializer argument, the seed is initialized once when the statement is prepared, prior to execution.

- With a nonconstant initializer argument (such as a column name), the seed is initialized with the value for each invocation of RAND().

Reply:
Hi, AndreMouche. I checked the RandWithSeed implementation in tidb, seems it also cannot separate constant initializer argument and nonconstant initializer argument. The seed will also be initialized on every invocation.

Reply:
@AndreMouche The problem is I can't find a place here like the struct in TiDB
type builtinRandWithSeedSig struct {
	baseBuiltinFunc
	randGen *rand.Rand
}
to store the initialized random generator.

Reply:
can we store the Rand in the EvalContext so that we can pass a customized Rand for the test?

Reply:
We cannot store the `Rand` in the `EvalContext` since all expressions share one `EvalContext` @siddontang . Maybe we can store the `Rand` as an option in the struct `ScalarFunc` @xiangyuf ?

Reply:
@AndreMouche seems not feasible, fields in ScalarFunc should implement `std::fmt::Debug`;  crate rand = "0.3" has not implemented this.

Reply:
@AndreMouche hi, any suggestions here?

Reply:
any update @AndreMouche 

Seem there is no way for us to pass a Rand explicitly, I am fine with this PR now.

Reply:
Could you please implement `Debug` for `ScalarFunc` to fix it @xiangyuf ? I do not think it's a good idea to make the scalar functions differently with `tidb` @siddontang .

Reply:
@AndreMouche @siddontang I did some work around to store the rand in ScalarFunc. 1, Use CusRng to include the rand and implement Debug, PartialEq trait; 2, Initialized the rand of CusRng in Expression::build according to ScalarFuncSig; 3, used the RefCell to pass the mutable reference of rand to the impl function. Pls let me know if this is the proper way since I'm still new to the Ownership and Borrowing of rust.

Reply:
Done